src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessStartInfo.cs

```diff
@@ -55,6 +55,20 @@ public ProcessStartInfo(string fileName, string arguments)
             _arguments = arguments;
         }
 
+        /// <devdoc>
+        ///     Specifies the name of the application that is to be started, as well as a set
+        ///     of command line arguments to pass to the application.
+        /// </devdoc>
+        public ProcessStartInfo(string fileName, IEnumerable<string> arguments)
+        {
+            _fileName = fileName;
+
+            if(arguments != null)
```

Comment:
I can move these checks into the new `ProcessStartInfo` ctor

https://github.com/dotnet/runtime/blob/af1171b7c9ba985c93141b107f64e5dd31ab2d37/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs#L1328-L1331

I'm a little worried about consistency. The other ctors for `ProcessStartInfo` are not throwing `ArgumentNullException` and `Process.Start(string fileName, IEnumerable<string> arguments)` is the only overload of that method that throws.

Reply:
> The other ctors for ProcessStartInfo are not throwing

This is true, but they are old and we can't easily change their behavior. For the new APIs, we either allow for nulls by making an argument nullable, or we don't and throw. Since the API was not [approved](https://github.com/dotnet/runtime/issues/66450#issuecomment-1544500845) as nullable (no **?**), we should throw. By doing that we send a clear message in the API contract: nulls are not allowed, if you have a `null` arguments, just don't use this API.